key-events.go

```diff
@@ -41,48 +40,14 @@ func (j keyEventJSONMessageCodec) DecodeMessage(binaryMessage []byte) (message i
 	return message, err
 }
 
-type keyEventMessage struct {
-	Toolkit   string `json:"toolkit"`
-	KeyCode   int    `json:"keyCode"`
-	Type      string `json:"type"`
-	ScanCode  int    `json:"scanCode"`
-	Modifiers int    `json:"modifiers"`
-	Keymap    string `json:"keymap"`
-}
-
-// Flutter only support keydown & keyup events type.
-// 2019-05-19, LOC of the flutter keyevent handler:
-// https://github.com/flutter/flutter/blob/7a4c33425ddd78c54aba07d86f3f9a4a0051769b/packages/flutter/lib/src/services/raw_keyboard.dart#L291-L298
-// On GLFW we are receiving glfw.Repeat events, those events are,
-// unknown to the Flutter Framework.
-// To stay consistent with other embedders, glfw.Repeat is "seen" as a
-// keydown event.
-// Comment about RawKeyEvent on others embedders:
-// https://github.com/go-flutter-desktop/go-flutter#issuecomment-494998771
 func (p *keyeventPlugin) sendKeyEvent(window *glfw.Window, key glfw.Key, scancode int, action glfw.Action, mods glfw.ModifierKey) {
-
-	var typeKey string
-	if action == glfw.Release {
-		typeKey = "keyup"
-	} else if action == glfw.Press {
-		typeKey = "keydown"
-	} else if action == glfw.Repeat {
-		typeKey = "keydown"
-	} else {
-		fmt.Printf("go-flutter: Unknown key event type: %v\n", action)
+	event, err := keyboard.Normalize(key, scancode, mods, action)
+	if err != nil {
+		fmt.Printf("go-flutter: failed to Normalize key event: %v", err)
 		return
 	}
 
-	event := keyEventMessage{
-		KeyCode:   int(key),
-		Keymap:    "linux",
-		Toolkit:   "glfw",
-		Type:      typeKey,
-		ScanCode:  scancode,
-		Modifiers: int(mods),
-	}
-	err := p.keyEventChannel.Send(event)
-	if err != nil {
+	if err := p.channel.Send(event); err != nil {
```

Comment:
I prefer to never inline the assignment to err unless it's a function that only fetches some information or performs a check. In this case, the call to Send is actually important for the function a a whole, not just for this error check, so it feels off to inline it. But that could just be personal preference.